packages/next/src/client/components/react-dev-overlay/internal/helpers/use-error-handler.ts

```diff
@@ -1,23 +1,14 @@
 import { useEffect } from 'react'
-
-import { isNextRouterError } from '../../../is-next-router-error'
 import { isHydrationError } from '../../../is-hydration-error'
 import { attachHydrationErrorState } from './attach-hydration-error-state'
 
 export type ErrorHandler = (error: Error) => void
 
-if (typeof window !== 'undefined') {
-  try {
-    // Increase the number of stack frames on the client
-    Error.stackTraceLimit = 50
-  } catch {}
-}
-
 let hasHydrationError = false
 const errorQueue: Array<Error> = []
-const rejectionQueue: Array<Error> = []
 const errorHandlers: Array<ErrorHandler> = []
-const rejectionHandlers: Array<ErrorHandler> = []
+export const rejectionQueue: Array<Error> = []
+export const rejectionHandlers: Array<ErrorHandler> = []
```

Comment:
feels strange to export these.. maybe they should be passed into `handleGlobalErrors`?

Reply:
I moved the definition back to avoid exports those states. Still keep the invocation in the separate file that we can import it like a side effect module into app-index to apply on the top of the imports